### PR TITLE
Feature/markup transactions

### PR DESCRIPTION
**Pull request by Dragonesis**

### CODE REVIEW

File: src/components/TransactionItem/Details/Details.js [hunk 1]
```diff
@@ -1,62 +1,146 @@
...
-    <div className={classNames('transaction-item-details', isActive && '-active')}>
-      <div className='item'>
-        <div className='label'>
-          <JText value='Tx hash' color='gray' />
+class TransactionItemDetails extends PureComponent<Props, StateProps> {
+  constructor(props: Props) {
+    super(props)
+
+    this.state = {
+      isMessage: false,
```
Comment (flexxnn) [inline src/components/TransactionItem/Details/Details.js]: Need to change isMessage to more readable name

File: src/components/TransactionItem/Details/Details.js [hunk 2]
```diff
@@ -1,62 +1,146 @@
...
+      assetDecimals,
+      assetSymbol,
+      isActive,
+    } = this.props
+
+    const {
+      isMessage,
+    }: StateProps = this.state
+
+    const fee = ((data.receiptData.gasUsed * data.data.gasPrice) / (10 ** assetDecimals))
```
Comment (flexxnn) [inline src/components/TransactionItem/Details/Details.js]: This calculation should use BigNumber

File: src/components/TransactionItem/Details/Details.js [hunk 3]
```diff
@@ -1,62 +1,146 @@
...
+  +assetDecimals: number,
+  +assetSymbol: string,
+  +isActive: boolean,
+|}
+
+type StateProps = {|
+  isMessage: boolean,
 |}
 
 const getTxLink = (txHash: Hash) => `${config.blockExplorerLink}/tx/${txHash}`
```
Comment (flexxnn) [inline src/components/TransactionItem/Details/Details.js]: We should use network Id to define txLink (to switch between ropsten and main)

File: src/components/TransactionItem/Details/Details.js [hunk 4]
```diff
@@ -1,62 +1,146 @@
...
+            <JText value='Tx hash' color='gray' />
+          </div>
+          <div className='value'>
+            <a
+              href={getTxLink(data.hash)}
+              target='_blank'
+              className='link'
+              rel='noopener noreferrer'
+            >
+              {data.hash}
```
Comment (flexxnn) [inline src/components/TransactionItem/Details/Details.js]: JText ?

File: src/components/TransactionItem/Details/Details.js [hunk 5]
```diff
@@ -1,62 +1,146 @@
...
+        <div className='actions'>
+          <div className='action' onClick={handle(repeat)(assoc('symbol', assetSymbol)(data))}>
+            <div className='icon'>
+              <JIcon size='medium' color='gray' name='repeat' />
+            </div>
+            <div className='text'>
+              <JText value='Repeat payment' color='gray' size='normal' />
+            </div>
+          </div>
+          <div className='action' onClick={addFavorite}>
```
Comment (flexxnn) [inline src/components/TransactionItem/Details/Details.js]: We should use one of our button component here, may be JRoundButton (as i remember)

File: src/components/TransactionItem/Details/Details.js (hunk 3 above)
Comment (IViol) [inline src/components/TransactionItem/Details/Details.js]: need to create `utils/transactions/getTxLink` and getAddressLink
they should have 2 args - value: string, networkId: NetworkId

File: src/components/TransactionItem/Main/Main.js [hunk 6]
```diff
@@ -1,77 +1,116 @@
...
 
-type Props = {
-  repeat: Function,
-  setActive: Function,
-  data: Transaction,
-  assetSymbol: ?string,
-  isActive: boolean,
-}
+type Props = {|
+  +setActive: Function,
```
Comment (flexxnn) [inline src/components/TransactionItem/Main/Main.js]: Function ?

File: src/components/TransactionItem/Details/Details.js (hunk 3 above)
Comment (IViol) [reply src/components/TransactionItem/Details/Details.js]: need to get subdomain for api from here:
https://github.com/jibrelnetwork/jwallet-web/blob/master/src/services/etherscan.js#L9-L14
this map should be placed in src/config.js for re-usage

File: src/components/TransactionItem/Main/Main.js [hunk 7]
```diff
@@ -1,77 +1,116 @@
...
+      assetDecimals,
+      assetSymbol,
+      isActive,
+      isCustom,
+      isReceived,
+    } = this.props
 
-function TransactionItemMain({ repeat, setActive, data, assetSymbol, isActive }: Props) {
-  const color = (data.type === 'receive') ? 'blue' : 'gray'
+    const amount = data.amount / (10 ** assetDecimals)
```
Comment (flexxnn) [inline src/components/TransactionItem/Main/Main.js]: Use BigNumber here

File: src/components/TransactionItem/Main/Main.js (hunk 6 above)
Comment (flexxnn) [reply src/components/TransactionItem/Main/Main.js]: (hash: string) => void

File: src/components/TransactionItem/Details/Details.js (hunk 2 above)
Comment (IViol) [inline src/components/TransactionItem/Details/Details.js]: need to create utils/transactions/getTxFee method. And use it in jsx without creating of var

File: src/components/TransactionItem/Details/Details.js [hunk 8]
```diff
@@ -1,62 +1,146 @@
...
+    }: StateProps = this.state
+
+    const fee = ((data.receiptData.gasUsed * data.data.gasPrice) / (10 ** assetDecimals))
+
+    return (
+      <div className={classNames('transaction-item-details', isActive && '-active')}>
+        <div className='item'>
+          <div className='label'>
+            <JText value='Tx hash' color='gray' />
+          </div>
```
Comment (IViol) [inline src/components/TransactionItem/Details/Details.js]: Tx hash -> TX Hash

File: src/components/TransactionsList/TransactionsList.js [hunk 9]
```diff
@@ -53,13 +55,15 @@ class TransactionsList extends Component<Props, ComponentState> {
 
     return (
       <div className='transactions-list'>
+
```
Comment (flexxnn) [inline src/components/TransactionsList/TransactionsList.js]: Extra line?

File: src/routes/Transactions/TransactionsIndexView.js [hunk 10]
```diff
@@ -11,6 +11,7 @@ type Props = {|
   +changeSearchInput: (string) => void,
   +transactions: TransactionWithAssetAddress[],
```
Comment (flexxnn) [inline src/routes/Transactions/TransactionsIndexView.js]: Array<TransactionWithAssetAddress>

File: src/components/TransactionItem/Details/Details.js [hunk 11]
```diff
@@ -1,62 +1,146 @@
...
+  render() {
+    const {
+      addFavorite,
+      repeat,
+      data,
+      assetDecimals,
+      assetSymbol,
+      isActive,
+    } = this.props
+
```
Comment (IViol) [inline src/components/TransactionItem/Details/Details.js]: need to destruct all used fields from `data` prop

File: src/components/TransactionItem/Details/Details.js [hunk 12]
```diff
@@ -1,62 +1,146 @@
...
         </div>
-        <div className='value'>
-          <JText value={`${data.fee} ETH`} color='gray' />
+        <div className='item'>
+          <div className='label'>
+            <JText value='Fee' color='gray' />
+          </div>
+          <div className='value'>
+            <JText value={`${fee} ${assetSymbol}`} color='gray' />
+          </div>
```
Comment (IViol) [inline src/components/TransactionItem/Details/Details.js]: `${fee} ${assetSymbol}` -> `${fee} ETH`

File: src/components/TransactionItem/Details/Details.js [hunk 13]
```diff
@@ -1,62 +1,146 @@
...
-            onClick={handle(repeat)(assoc('symbol', assetSymbol)(data))}
-            color='white'
-            label='Repeat'
-            iconColor='blue'
-            iconName='repeat'
-            labelColor='blue'
-          />
+        <div className='actions'>
+          <div className='action' onClick={handle(repeat)(assoc('symbol', assetSymbol)(data))}>
+            <div className='icon'>
```
Comment (IViol) [inline src/components/TransactionItem/Details/Details.js]: just `repeat` for handler

File: src/components/TransactionItem/Details/Details.js [hunk 14]
```diff
@@ -1,62 +1,146 @@
...
+  +addFavorite: Function,
+  +repeat: Function,
+  +data: TransactionWithAssetAddress,
+  +assetDecimals: number,
+  +assetSymbol: string,
+  +isActive: boolean,
+|}
+
+type StateProps = {|
+  isMessage: boolean,
```
Comment (IViol) [inline src/components/TransactionItem/Details/Details.js]: isMessage -> isCommenting

File: src/components/TransactionItem/Details/Details.js [hunk 15]
```diff
@@ -1,62 +1,146 @@
...
+            <div className='text'>
+              <JText value='Add comment' color='gray' size='normal' />
+            </div>
+          </div>
         </div>
+        {isMessage && (
+          <TransactionItemMessage
+            saveMessage={this.toggleMessage}
+            deleteMessage={this.toggleMessage}
+          />
```
Comment (IViol) [inline src/components/TransactionItem/Details/Details.js]: TransactionItemMessage -> TransactionItemDetailsComment

Comment (IViol) [reply src/components/TransactionItem/Details/Details.js]: saveComment, etc

File: src/components/TransactionItem/Details/Message/Message.js [hunk 16]
```diff
@@ -0,0 +1,50 @@
...
+
+    return (
+      <div className='transaction-item-message'>
+        <label htmlFor='message' className='icon'>
+          <JIcon size='medium' color='gray' name='message-add' />
+        </label>
+
+        <input type='text' id='message' className='field' placeholder='Your comment' />
+
+        <div className='actions'>
```
Comment (IViol) [inline src/components/TransactionItem/Details/Message/Message.js]: remove empty line please

File: src/components/TransactionsList/TransactionsList.js [hunk 17]
```diff
@@ -53,13 +55,15 @@ class TransactionsList extends Component<Props, ComponentState> {
...
         {items.map((item: TransactionWithAssetAddress) => (
           <TransactionItem
             key={item.hash}
             setActive={this.setActive(item.hash)}
             data={item}
             asset={digitalAssets[item.assetAddress]}
             isActive={activeItems.includes(item.hash)}
+            isReceived={ownerAddress.toLowerCase() === item.from}
```
Comment (IViol) [inline src/components/TransactionsList/TransactionsList.js]: ownerAddress.toLowerCase() === item.from.toLowerCase()

File: src/components/TransactionItem/Details/Details.js (hunk 5 above)
Comment (IViol) [reply src/components/TransactionItem/Details/Details.js]: JFlatButton

File: src/components/TransactionItem/Main/Main.js (hunk 7 above)
Comment (IViol) [reply src/components/TransactionItem/Main/Main.js]: need to create getTxAmount util method

File: src/components/TransactionItem/Details/Details.js [hunk 18]
```diff
@@ -20,7 +20,8 @@ type StateProps = {|
   isCommenting: boolean,
 |}
 
-const getTxLink = (txHash: Hash) => `${config.blockExplorerLink}/tx/${txHash}`
+// const getTxLinks = (txHash: Hash) => getTxLink(txHash, networkId)
+// console.warn(getTxLinks)
```
Comment (IViol) [inline src/components/TransactionItem/Details/Details.js]: remove comments

File: src/components/TransactionItem/Details/Details.js [hunk 19]
```diff
@@ -71,12 +78,12 @@ class TransactionItemDetails extends PureComponent<Props, StateProps> {
           </div>
           <div className='value'>
             <a
-              href={getTxLink(data.from)}
+              href={getTxLink(from, networkId)}
```
Comment (IViol) [inline src/components/TransactionItem/Details/Details.js]: it should be address link

File: src/components/TransactionItem/Details/Details.js [hunk 20]
```diff
@@ -71,12 +78,12 @@ class TransactionItemDetails extends PureComponent<Props, StateProps> {
...
               target='_blank'
               className='link'
               rel='noopener noreferrer'
             >
-              <JText value={data.from} color='blue' />
+              <JText value={from} color='blue' weight='bold' />
```
Comment (IViol) [inline src/components/TransactionItem/Details/Details.js]: it should be address link

File: src/components/TransactionItem/Details/Details.js [hunk 21]
```diff
@@ -86,12 +93,12 @@ class TransactionItemDetails extends PureComponent<Props, StateProps> {
           </div>
           <div className='value'>
             <a
-              href={getTxLink(data.to)}
+              href={getTxLink(to, networkId)}
               target='_blank'
```
Comment (IViol) [inline src/components/TransactionItem/Details/Details.js]: it should be address link

File: src/components/TransactionItem/Main/Main.js [hunk 22]
```diff
@@ -50,12 +55,19 @@ class TransactionItemMain extends PureComponent<Props> {
...
             <div className='hash'>
-              <JText
-                value={isReceived ? data.to : data.from}
-                color={color}
-                weight='bold'
-                size='normal'
-              />
+              <a
+                href={getTxLink(isReceived ? data.to : data.from, networkId)}
+                target='_blank'
```
Comment (IViol) [inline src/components/TransactionItem/Main/Main.js]: getAddressLink

**Pull request merged**